lib/metasploit/framework/data_service/remote/http/remote_loot_data_service.rb

```diff
@@ -8,16 +8,22 @@ module RemoteLootDataService
 
   def loot(opts = {})
     path = get_path_select(opts, LOOT_API_PATH)
-    # TODO: Add an option to toggle whether the file data is returned or not
-    loots = json_to_mdm_object(self.get_data(path, nil, opts), LOOT_MDM_CLASS, [])
-    # Save a local copy of the file
-    loots.each do |loot|
-      if loot.data
-        local_path = File.join(Msf::Config.loot_directory, File.basename(loot.path))
-        loot.path = process_file(loot.data, local_path)
+    data = self.get_data(path, nil, opts)
+    rv = json_to_mdm_object(data, LOOT_MDM_CLASS, [])
+    parsed_body = JSON.parse(data.response.body, symbolize_names: true)
+    data = parsed_body[:data]
+    data.each do |loot|
+      # TODO: Add an option to toggle whether the file data is returned or not
+      if loot[:data]
+        local_path = File.join(Msf::Config.loot_directory, File.basename(loot[:path]))
+        rv[data.index(loot)].path = process_file(loot[:data], local_path)
+      end
+      if loot[:host]
+        host_object = to_ar(RemoteHostDataService::HOST_MDM_CLASS.constantize, loot[:host])
```

Comment:
What is the purpose of calling `constantize` on `HOST_MDM_CLASS` when this isn't performed earlier on `LOOT_MDM_CLASS` or in other locations?

Reply:
The other methods are calling `json_to_mdm_object`, this is calling `to_ar` directly. `json_to_mdm_object` does the `constantize` call within it when it calls `to_ar`.